### DISCUSSION TITLE
该怎么看TVM和XLA？

### DISCUSSION THREAD

**Discussion by kun-teng** [mlc-ai/mlc-zh#7] [General]
新手: 学新东西时，习惯比较相关概念。
好奇：该怎么看TVM和XLA?
https://github.com/google/jax/issues/10277

**Comment by yzh119** [comment]
HLO是XLA的IR，用于表示计算图级别的语义，可以用来进行算子融合等优化，支持自家的TPU及其它硬件。
HLO对应于TVM的[Relay IR](https://arxiv.org/pdf/1904.08368.pdf)（后续工作为[Relax IR](https://github.com/tlc-pack/relax)），TVM还有一层算子级别的IR：Tensor IR，可以进行自定义算子及自动调度优化。统一这两层的IR表示是TVM接下来的工作重心之一。

开源社区的项目都会持续进化并互相借鉴，所以这些论断可能半年之后就会变化，建议不要纠结于概念。

**Reply by HLearning** [reply]
统一两层IR， 说的是relax吧？
